### PR TITLE
fix(admin-ui, admin): Use Webpack for building admin UI 

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Removes Vite from the project, in favor of Webpack.
- Removes packages `@medusajs/admin-shared` and `@medusajs/admin-sdk`.
- Changes `medusa develop` command to automatically start up an admin development server if `@medusajs/admin` is installed.
- Updates `@medusajs/admin` setup loader to automatically build admin UI, if the last build is outdated (determined by changes to dependencies, local admin code, plugin options, plugins in `medusa-config.js`), if `autoRebuild` is enabled and the initiating command is `medusa start`.
- Adds support for custom webpack configurations, which can be useful, if a developer needs to alias a dependency of their extensions, load custom CSS, etc.

**Why**
- Vite was dropped due to it not being flexible enough to achieve our desired setup for running a local admin development server for extension work.

**Changes to existing code**
- Add support for `.before` and `.after` for all domain `list` and `domain` injection zones.

**Changes to working with admin extensions**
- To inject a `widget` it needs to be placed in `src/admin/widgets` in order for it to be validated.
- `***WidgetProps` types are now exported from `@medusajs/admin`, the same applies to `WidgetConfig` **(Notice the name change from `ExtensionsConfig` -> `WidgetConfig`, and that you should no longer specify a `type`)**.
- To add custom webpack configurations, you can create a file at `/src/admin/webpack.config.js`, and use the `withCustomWebpack` function to override or add to the existing config:

```js
const { withCustomWebpackConfig } = require("@medusajs/admin/lib");

module.exports = withCustomWebpackConfig((config) => {
  return {
    ...config,
    resolve: {
      ...config.resolve,
      alias: {
        ...config.resolve.alias,
        "my-dependency$": require.resolve("my-dependency), 
      },
    },
  };
});
```

Closes CORE-1338, CORE-1340

### CODE REVIEW

File: packages/ui/tailwind.content.js
Comment (adrien2p) [inline packages/ui/tailwind.content.js:3]: question: what is my-medusa-store? in that context?

File: packages/admin-ui/tsconfig.json
Comment (adrien2p) [inline packages/admin-ui/tsconfig.json]: question: Since the root dir is src and we are including other files from the parent, will the dist  looks like
- dist
  - src
  - vite
  - tsup

**Review by adrien2p — APPROVED** [review]
LGTM, just small questions

**Comment by kasperkristensen** [comment]
Sorry @adrien2p forgot to set it as a draft, no need to review this right now, was testing out if I could fix using Vite for the dev command

**Comment by adrien2p** [comment]
> Sorry @adrien2p forgot to set it as a draft, no need to review this right now, was testing out if I could fix using Vite for the dev command

ahah my bad, sorry @kasperkristensen

File: packages/admin-ui/src/node/actions/clean.ts
Comment (olivermrbl) [inline packages/admin-ui/src/node/actions/clean.ts]: question: This meant to be here?

File: packages/admin-ui/src/node/webpack/get-custom-webpack-config.ts
Comment (olivermrbl) [inline packages/admin-ui/src/node/webpack/get-custom-webpack-config.ts]:
```suggestion
```

File: packages/admin-ui/ui/src/domain/collections/details/index.tsx
Comment (olivermrbl) [inline packages/admin-ui/ui/src/domain/collections/details/index.tsx]: thought: Should this not be `before`?
```suggestion
          {getWidgets("product_collection.details.before").map((w, i) => {
            return (
              <WidgetContainer
                key={i}
                entity={collection}
                injectionZone="product_collection.details.before"
```

**Review by olivermrbl — CHANGES_REQUESTED** [review]
lets gooooo! insane piece of work @kasperkristensen 🚀 🚀 

minor things ... also, let's remove the changesets for the admin packages and figure out a beta release path for those

**Comment by kasperkristensen** [comment]
> lets gooooo! insane piece of work @kasperkristensen 🚀 🚀
> 
> minor things ... also, let's remove the changesets for the admin packages and figure out a beta release path for those

Yea let's remove all the changesets, but let us wait until we have merged this into the feature branch, as creating snapshots gets a little weird when their are no changesets for the packages 💪

**Pull request merged**
